### DISCUSSION TITLE
When use the Trace&log Kafka exporter functions, I cann't see skywalking producer create relevant topics successful.

### DISCUSSION THREAD

**Discussion by Henry75m39** [apache/skywalking#11943] [Q&A] [answered]
I'm trying the version 9.1.0 skywalking java agent to trace my springboot apps.  The Trace &Log Kafka Exporter are provide the capability to forward log data  to othere system.  However, I have encountered one blocker when I enabled this function, I cannot see any message come into the topics.

- 1. I have enabled my skywalking configuration, like this:
![image](https://github.com/apache/skywalking/assets/42402567/7ff813ce-cf5f-48ea-be30-af59bbe6cbf8)
- 2. After above configuration, restart skywalking
- 3. Change the local java agent config file, just fill up the server address and port number, like this:
![image](https://github.com/apache/skywalking/assets/42402567/518ebf85-fc6a-44b8-8e0d-1bbe4ebc5432)
- 4. Add kafka config in application.properties file, like this:
![image](https://github.com/apache/skywalking/assets/42402567/1cb53555-346d-41c1-8b13-771954e267ec)
- 5. Write my consumer client code, like this:
![image](https://github.com/apache/skywalking/assets/42402567/8df4b926-c362-41df-bb94-2af30f6ec52a)
- 6. Start my boot apps, and login my system, then click some functions on the web frontend to monitor the kafka group and topics. like this:
![image](https://github.com/apache/skywalking/assets/42402567/b46f1f54-a52c-42f0-9183-55d03cd832a7)

The current situation is, I cannot see any content in the topic of 'skywalking-trace' or 'skywalking-log',  I was tried to change the topic to 'skywalking-export-trace' or 'skywalking-log'.  Therefore, follow the doc, I can't what I want.  I'm asking help to figure out what's the problem.
https://skywalking.apache.org/docs/main/next/en/setup/backend/exporter/#trace-kafka-exporter

**Comment by wu-sheng** [comment]
What do you want to do? The exporter is at OAP server side, why agent config is involved? I am not following.

**Reply by wu-sheng** [reply]
You seems not to read documents correctly. If you want communicationz between agent and OAP through Kafka, OAP should active Kafka receiver, rather than exporter.

Exporter is forwarding processed data from OAP to 3rd party system.

**Reply by Henry75m39** [reply]
Thanks Wu, I just want to use the Trace&Log Kafka exporter to forward data from related topics to the other systems. But the current situation is, I cannot see any content in the topic of 'skywalking-trace' or 'skywalking-log after enabling this function, I was tried to change the topic to 'skywalking-export-trace' or 'skywalking-log', and got the same result.  I was follow this doc link to do the configuration. 
My question here: 
- a. To active exporter function, do I need to change the kafka related section in my java agent configuration?
- b. To make the exporter work, do I need move the 'kafka-reporter-plugin-9.1.0.jar' from optional-reporter-plugins to plugins folder?
- c. in the offical link of expoter ( https://skywalking.apache.org/docs/main/next/en/setup/backend/exporter/#trace-kafka-exporter), the topics name are record as 'skywalking-trace' & 'skywalking-log', but in OAP server application.yaml, the topics name are different, they are record as 'skywalking-export-trace'  & 'skywalking-export-log', my question which topic name should I use?

**Reply by wu-sheng** [reply]
> a. To active exporter function, do I need to change the kafka related section in my java agent configuration?

No, that is irrelevant. 

> b. To make the exporter work, do I need move the 'kafka-reporter-plugin-9.1.0.jar' from optional-reporter-plugins to plugins folder?

No, that is irrelevant, too. The exporter is working separated from the agent, don't consider agent as a part of it.

> c. in the offical link of expoter ( https://skywalking.apache.org/docs/main/next/en/setup/backend/exporter/#trace-kafka-exporter), the topics name are record as 'skywalking-trace' & 'skywalking-log', but in OAP server application.yaml, the topics name are different, they are record as 'skywalking-export-trace' & 'skywalking-export-log', my question which topic name should I use?

I think we should follow the `application.yml`.

```
    kafkaTopicTrace: ${SW_EXPORTER_KAFKA_TOPIC_TRACE:skywalking-export-trace}
    kafkaTopicLog: ${SW_EXPORTER_KAFKA_TOPIC_LOG:skywalking-export-log}
```

FYI @wankai123 Is this a doc error?

**Reply by Henry75m39** [reply]
@wu-sheng Hi bro, thanks for your guidance, I changed all configuration back to the orginal. And use the skywalking-export-trace topic to redo the test my program.  I still don't see any content in that topic, pls see the screenshot. 

![image](https://github.com/apache/skywalking/assets/42402567/c7df9c56-e2db-44dc-942e-a20558911483)

my skywalking OAP server config as:
![image](https://github.com/apache/skywalking/assets/42402567/adefc6b6-d0ee-4a4d-b228-611491f5bc27)

Any other suggestion or idea for me to fix this blocker? Many thx.

**Reply by wu-sheng** [reply, marked as answer]
You missed one important part, `selector` need to be set as `default` to activate.

**Reply by Henry75m39** [reply]
very appreicated your quick response, I fix it finally after set selector as default
![image](https://github.com/apache/skywalking/assets/42402567/39c9c597-8765-4775-aac8-6f1afea174a1)

I would like to suggest update the instratution on the offical website.  
https://skywalking.apache.org/docs/main/next/en/setup/backend/exporter/#trace-kafka-exporter. 
There are 2 key information should be documented:
- 1.  The topic names should be corrected to "skywalking-export-trace" and "skywalking-export-log" to align with the yml definition.
- 2.  To activate the exporter, you should set selector to default and set ${SW_EXPORTER_ENABLE_KAFKA_LOG:true} in exporter section.

**Reply by wu-sheng** [reply]
I think the first has been fixed today morning.

**Reply by wu-sheng** [reply]
You could send a pull request to update this file, https://github.com/apache/skywalking/blob/master/docs/en/setup/backend/exporter.md

I will review that.

**Reply by Henry75m39** [reply]
already committed a change request. hope my effort can help skywalking getting better.

**Reply by wu-sheng** [reply]
Fixed by https://github.com/apache/skywalking/pull/11951

**Reply by Superskyyy** [reply]
@Henry75m39 Thank you for your contribution!
